source/notes/openstack.rst

```diff
@@ -96,3 +96,64 @@ You have to populate the database.
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server   File "/var/lib/kolla/venv/local/lib/python2.7/site-packages/oslo_messaging/rpc/server.py", line 163, in _process_incoming
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server     res = self.dispatcher.dispatch(message)
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server   File "/var/lib/kolla/venv/local/lib/python2.7/site-packages/oslo_messaging/rpc/dispatcher.py", line 265, in dispatch
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server     return self._do_dispatch(endpoint, method, ctxt, args)
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server   File "/var/lib/kolla/venv/local/lib/python2.7/site-packages/oslo_messaging/rpc/dispatcher.py", line 194, in _do_dispatch
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server     result = func(ctxt, **new_args)
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server   File "/var/lib/kolla/venv/local/lib/python2.7/site-packages/nova/conductor/manager.py", line 1396, in schedule_and_build_instances
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server     limits=host.limits, host_list=host_list)
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server   File "/var/lib/kolla/venv/local/lib/python2.7/site-packages/nova/compute/rpcapi.py", line 1064, in build_and_run_instance
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server     cctxt.cast(ctxt, 'build_and_run_instance', **kwargs)
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server   File "/var/lib/kolla/venv/local/lib/python2.7/site-packages/oslo_messaging/rpc/client.py", line 151, in cast
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server     self._check_version_cap(msg.get('version'))
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server   File "/var/lib/kolla/venv/local/lib/python2.7/site-packages/oslo_messaging/rpc/client.py", line 128, in _check_version_cap
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server     version_cap=self.version_cap)
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server RPCVersionCapError: Requested message version, 5.0 is incompatible.  It needs to be equal in major version and less than or equal in minor version as the specified version cap 4.17.
+   12:05:33.840 35 ERROR oslo_messaging.rpc.server
+
+To verify your problem, take a look inside the `nova` database at the `services` table.
+You can find the service version in the `version` column.
+
+Output before and after removing old disabled `nova-compute` services:
+
+.. code-block:: console
+
+   nova-consoleauth.log: 12:38:01.998 7 INFO nova.compute.rpcapi [req-7bc75c8d-a3c2-4961-81d3-91e2cfe2f382 - - - - -] Automatically selected compute RPC version 4.17 from minimum service version 22
+   ...
+   $ openstack compute service delete...
+   ...
+   nova-consoleauth.log: 13:43:15.488 7 INFO nova.compute.rpcapi [req-48feeaab-63f0-44a7-b2fe-90134ec61d82 - - - - -] Automatically selected compute RPC version 5.0 from minimum service version 35
+
+Solution
```

Comment:
This an the following content should be placed in line 99...

Reply:
Not sure if this is resolved with my last push. Please review again.